### DISCUSSION TITLE
[v7] useController Typescript

### DISCUSSION THREAD

**Discussion by writes-of-spring** [react-hook-form/react-hook-form#4423] [Q&A] [answered]
Hey, really looking forward to v7!

Question about using useController with Typescript, with an input created using use controller I get this typescript error Type 'Record<string, any>' is not assignable to type 'FormValues'. Will I need to create a useController input per form going forward? 

codesandbox link https://codesandbox.io/s/usecontroller-forked-hwree?file=/src/App.tsx

Thanks!

**Comment by bluebill1049** [comment, marked as answer]
`UseControllerProps<FormValues>` need to pass down the generic

**Reply by writes-of-spring** [reply]
I had never gotten that to work before, but I think I got it, thanks a bunch!!

This is what I ended up with for a reusable mui textfield:

```js 
const Input = <FormType extends object>({
  control,
  name,
  rules = {},
  ...otherProps
}: InputProps<FormType>) => {
  const {
    field: { ref, ...inputProps },
    fieldState: { invalid, isTouched, isDirty },
  } = useController<FormType>({
    name,
    control,
    rules,
    defaultValue: "",
  });

  return (
    <TextField {...inputProps} {...otherProps} inputRef={ref} error={invalid} />
  );
};

```
